esp-hal-common/src/lib.rs

```diff
@@ -73,6 +73,12 @@ pub mod system;
 
 pub mod analog;
 
+#[cfg_attr(feature = "esp32", path = "cpu_control/esp32.rs")]
+#[cfg_attr(feature = "esp32c3", path = "cpu_control/none.rs")]
+#[cfg_attr(feature = "esp32s2", path = "cpu_control/none.rs")]
+#[cfg_attr(feature = "esp32s3", path = "cpu_control/esp32s3.rs")]
+pub mod cpu_control;
```

Comment:
```suggestion
#[cfg_attr(feature = "esp32", path = "cpu_control/esp32.rs")]
#[cfg_attr(feature = "esp32s3", path = "cpu_control/esp32s3.rs")]
pub mod cpu_control;

#[cfg(any(feature = "esp32c3", feature = "esp32s2"))]
pub mod cpu_control {}
```

Probably nitpicking as usual (😁) but maybe something like this would be better? Seems kind of strange to require an empty source file to me, but also whatever works is fine at the end of the day.

Reply:
Yeah - unfortunately the compiler hates it this way 😆 so I think we need to keep the `none.rs`